test/widgets/dialog_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/dialog.dart';
+
+import '../model/binding.dart';
+import 'dialog_checks.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  late BuildContext context;
+
+  const String title = "Dialog Title";
+  const String message = "Dialog message.";
```

Comment:
nit: omit types when they're obvious:
```suggestion
  const title = "Dialog Title";
  const message = "Dialog message.";
```